feat(uv): migrated the rigel project from poetry to uv

robotairio opened the pull request:
Closes #33 

- feat(uv): migrated from poetry to uv
- feat(ci): migrated CI pipelines from poetry to uv

MisterOwlPT reviewed (changes requested):
Hi @robotairio!
Thanks for the very cool PR!

Some points and we are ready to merge:
- Create a `Makefile` with a different target for each `uv run ...` command called in the CI pipeline and replace them with calls to `make ...`. The idea here is to have all commands defined in a single place and with a same invocation for both local and cloud environments.

- The existing `noxfile.py` was not updated. Was this intentional? How can we test locally that Rigel is working for all the same target Python the same as we do within the CI configuration file? Consider also listing `nox` as a development dependency and perhaps create a `make` to invoke it.

Let me know what you think about this.

CC @rarrais

robotairio commented:
Changes
- Added a Makefile with targets for lint, typecheck, test, build, publish, and nox.
- Updated CI to use Makefile targets for all commands.
- Updated noxfile.py to use uv instead of poetry.
- Migrated to ruff from flake8 as it was painfully slow
- Applied pre-commit hooks to the repo

@MisterOwlPT While running the noxfile, I found this

`E   ImportError: cannot import name 'Package' from 'rigel.models.application'`
We have test files trying to import `Package `or `SSHKey`  `from rigel.models.application`, but these symbols are not present there.

MisterOwlPT commented:
Hi @robotairio 

> We have test files trying to import Package or SSHKey from rigel.models.application, but these symbols are not present there.

Yes, they were deleted when the support for container secrets was added.
The tests were not updated, unfortunately (but there's an issue already for that).

---

Thanks a lot for your contributions!
We are ready to merge 😄
